src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs

```diff
@@ -199,19 +200,55 @@ public bool IsConnected
             }
         }
 
+        public static IPEndPoint GetHandshakedEndPoint(INetworkPeer peer)
+        {
+            if (peer.Inbound && peer.State == NetworkPeerState.HandShaked)
+            {
+                IPEndPoint addressFrom = peer.PeerVersion?.AddressFrom.MapToIpv6();
+
+                if (addressFrom != null && !addressFrom.Address.Equals(IPAddress.IPv6Any))
```

Comment:
@quantumagi Just trying to understand this... because you now mapped `peer.PeerVersion?.AddressFrom` to IPv6 you are implying that this (from the previous code):

```!peer.PeerVersion.AddressFrom.Match(new IPEndPoint(IPAddress.Loopback, this.AttachedPeer.Network.DefaultPort))```

Is the same check as:

```!addressFrom.Address.Equals(IPAddress.IPv6Any)```

Reply:
As mentioned above:
>The use of this.AttachedPeer.Network.DefaultPort may be problematic and is removed. The reason it is problematic is in the use of the default network port. There is no reason that would be guaranteed to be right as default ports can be overridden.

The `!addressFrom.Address.Equals(IPAddress.IPv6Any)` is added to disregard `AddressFrom` addresses deriving from `0.0.0.0` due to those addresses usually also having port = 0. I.e. we don't want to use that address or the associated port number.